### PR TITLE
Adding Ara columns for Composition values

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
This PR adds four column definitions for composition-valued variables in Ara definitions.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: src/citrine/ara/columns.py [hunk 1]
```diff
@@ -194,6 +209,111 @@ def __init__(self, *,
...
+
+class NthBiggestComponentNameColumn(Serializable["NthBiggestComponentNameColumn"], Column):
+    """[ALPHA] Name of the Nth biggest component.
+
+    Parameters
+    ----------
+    data_source: str
+        name of the variable to use when populating the column
+    n: int
+        index of the component name to extract, starting with 1 for the biggest
```
Comment (bfolie) [inline src/citrine/ara/columns.py]: What happens if `n` is larger than the number of components? For `NthBiggestComponentQuantityColumn` I would expect to get a value of 0, but I do not know what to expect here.

File: src/citrine/ara/columns.py [hunk 2]
```diff
@@ -194,6 +209,111 @@ def __init__(self, *,
...
+    data_source = properties.String('data_source')
+    n = properties.Integer("n")
+    typ = properties.String('type', default="biggest_component_name_column", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["data_source", "n", "typ"]
+
+    def __init__(self, *,
+                 data_source: str,
+                 n: int):
```
Comment (bfolie) [inline src/citrine/ara/columns.py]: Should we set `n=1` as the default? I think it depends on how we expect people to use this column (will they almost always be grabbing the largest amount?)

Comment (maxhutch) [reply src/citrine/ara/columns.py]: Given that "N" is in the name, I'd say "no".

File: src/citrine/ara/columns.py (hunk 1 above)
Comment (maxhutch) [reply src/citrine/ara/columns.py]: Its empty.  I'll clarify in docs.

**Pull request merged**